### PR TITLE
MyStudies Response Server UI

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
Consolidate UI using more modern technologies to improve user experience.

#### Related Pull Requests
* N/A

#### Changes
* Rename folder management tab from Response Forwarding to MyStudies Response Server
* Rename webpart from MyStudies Study Setup to MyStudies Response Server
* Add 'foldersWithStudyIdConfiguredCount' metric
* Migrate settings on legacy 'Study Setup' webpart to current MyStudies Response Server Folder Management Tab
* Set up read-only MyStudies Response Server webpart
* Update tests

### CODE REVIEW

File: test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java [hunk 1]
```diff
@@ -62,8 +63,6 @@ protected void doCleanup(boolean afterTest)
     @Test
     public void testStudyName()
     {
-        final String PROMPT_NOT_ASSIGNED = "Enter the StudyId to be associated with this folder. The StudyId should be the same as it appears in the study design interface.";
```
Comment (RosalineP) [inline test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java:65]: We have removed this upper text from the UI.

File: test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java [hunk 2]
```diff
@@ -75,57 +74,44 @@ public void testStudyName()
...
-        setupPage.getStudySetupWebPart().clickSubmit();
-
-        log("Validate that the submit button is disabled after you click it.");
-        assertFalse("Submit button is showing as enabled, it should not be.", setupPage.getStudySetupWebPart().isSubmitEnabled());
+        forwardingTab.setInputId(STUDY_NAME01);
+        forwardingTab.validateSaveButtonEnabled();
+        forwardingTab.saveAndExpectSuccess();
 
-        log("Remove the web part,bring it back and validate the study name is still there.");
-        setupPage.getStudySetupWebPart().remove();
```
Comment (RosalineP) [inline test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java:93]: We have removed the web part in favor of adding the UI to the MyStudies Response Server  Folder Management tab.

File: test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java [hunk 3]
```diff
@@ -153,10 +139,6 @@ public void testStudyName()
         // Get a new setupPage.
         setupPage = new SetupPage(this);
         validateGridInfo(setupPage, batchId, expectedTokenCount, Integer.toString(tokensToAssign.size()));
-
-        log("Validate the prompt.");
-        assertEquals("The prompt is not as expected.", PROMPT_ASSIGNED.replace("$STUDY_NAME$", STUDY_NAME01.toUpperCase()), setupPage.getStudySetupWebPart().getPrompt());
```
Comment (RosalineP) [inline test/src/org/labkey/test/tests/response/ConfigAndEnrollTest.java:158]: We have removed the upper prompt, as mentioned, and altered the whacky UI such that the name field is always visible.

File: test/src/org/labkey/test/tests/response/TokenValidationTest.java [hunk 4]
```diff
@@ -227,20 +231,22 @@ public void testMyStudiesCoordinatorRole()
...
-        setupPage.validateSubmitButtonDisabled();  // Submit button should be present
+        ForwardingTab forwardingTab = ForwardingTab.beginAt(this);
+        forwardingTab.validateSaveButtonDisabled();  // Save button should be present
+        SetupPage.beginAt(this, PROJECT_NAME01);
 
         // Test for Reader
         impersonateRole("Reader");
         assertTrue(batchesWebPart.isNewBatchPresent());
         assertFalse(batchesWebPart.isNewBatchEnabled());
-        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
```
Comment (RosalineP) [inline test/src/org/labkey/test/tests/response/TokenValidationTest.java:236]: UI is now located in Folder Management, which a Reader will not have access to

File: test/src/org/labkey/test/tests/response/TokenValidationTest.java [hunk 5]
```diff
@@ -227,20 +231,22 @@ public void testMyStudiesCoordinatorRole()
...
+        ForwardingTab.beginAt(this);
+        assertTextPresent("User does not have permission to perform this operation.");  // Not authorized
         stopImpersonating(false);
 
         // Test for MyStudies Coordinator
         impersonateRoles("Reader", "MyStudies Coordinator");
         assertTrue(batchesWebPart.isNewBatchPresent());
         assertTrue(batchesWebPart.isNewBatchEnabled());  // Should be able to create a new batch
-        assertFalse(setupPage.isSubmitButtonVisible());  // Submit button should NOT be present
```
Comment (RosalineP) [inline test/src/org/labkey/test/tests/response/TokenValidationTest.java:243]: UI is now located in Folder Management, which a MyStudies Coordinator will not have access to

**Review by labkey-tchad — DISMISSED** [review]

**Comment by RosalineP** [comment]
Did a name change that broke tests as of last commit—fixing it up now

**Review by labkey-tchad — APPROVED** [review]
Approved assuming tests pass

File: resources/web/mobileAppStudy/panel/studySetup.js [hunk 6]
```diff
@@ -1,290 +0,0 @@
-/*
```
Comment (RosalineP) [inline resources/web/mobileAppStudy/panel/studySetup.js:1]: Not too familiar with ext architecture—is there more I ought to delete?

File: src/org/labkey/response/ResponseController.java [hunk 7]
```diff
@@ -977,7 +974,7 @@ public void setActivityVersion(String activityVersion)
         @Override
         protected HttpView getTabView(ForwardingSettingsForm form, boolean reshow, BindException errors)
         {
-            return new JspView<>("/org/labkey/response/view/forwarderSettings.jsp", form, errors);
+            return new JspView<>("/org/labkey/response/view/MyStudiesResponseServerSettings.jsp", form, errors);
```
Comment (labkey-adam) [inline src/org/labkey/response/ResponseController.java:977]: JSP files usually start with lowercase (by convention). Not worth changing, just FYI.

File: src/org/labkey/response/ResponseModule.java [hunk 8]
```diff
@@ -82,8 +86,8 @@ public boolean hasScripts()
     @NotNull
     protected Collection<WebPartFactory> createWebPartFactories()
     {
-        SimpleWebPartFactory studySetupFactory = new SimpleWebPartFactory("MyStudies Study Setup",
-            WebPartFactory.LOCATION_BODY, StudyConfigWebPart.class, null);
+        SimpleWebPartFactory studySetupFactory = new SimpleWebPartFactory("MyStudies Response Server",
+            WebPartFactory.LOCATION_BODY, MyStudiesResponseServerWebPart.class, null);
```
Comment (labkey-adam) [inline src/org/labkey/response/ResponseModule.java:90]: You should add a legacy name below, "MyStudies Study Setup". This maintains backward compatibility with existing webparts on the page or in a folder archive.

**Review by labkey-adam — CHANGES_REQUESTED** [review]
Only change is adding an alias for the webpart name

File: resources/web/mobileAppStudy/panel/studySetup.js (hunk 6 above)
Comment (labkey-adam) [reply resources/web/mobileAppStudy/panel/studySetup.js:1]: Would be worthwhile to get someone more familiar with JavaScript / Ext4 to do a quick review of the client code. Nick, Alan, Cory, etc.

File: src/org/labkey/response/ResponseModule.java (hunk 8 above)
Comment (RosalineP) [reply src/org/labkey/response/ResponseModule.java:90]: Resolved!

File: resources/web/mobileAppStudy/panel/studySetup.js (hunk 6 above)
Comment (RosalineP) [reply resources/web/mobileAppStudy/panel/studySetup.js:1]: Checked in with Nick and we're looking good 👍

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**